File: src/MicrosoftTeamsIntegration.Jira/Models/NotificationSubscriptionEvent.cs
using Newtonsoft.Json;

namespace MicrosoftTeamsIntegration.Jira.Models;

public class NotificationSubscriptionEvent
{
    [JsonProperty("subscription")]
    public NotificationSubscription Subscription { get; set; }

    [JsonProperty("action")]
    public SubscriptionAction Action { get; set; }
}

public enum SubscriptionAction
{
    Created,
    Updated,
    Deleted
}

File: src/MicrosoftTeamsIntegration.Jira/ClientApp/src/app/core/models/NotificationSubscriptionEvent.ts
import { NotificationSubscription } from './NotificationSubscription';

export interface NotificationSubscriptionEvent {
    subscription: NotificationSubscription;
    action: NotificationSubscriptionAction;
}

export enum NotificationSubscriptionAction {
    Create = 'Created',
    Update = 'Updated',
    Delete = 'Deleted',
    Enabled = 'Enabled',
    Disabled = 'Disabled',
}
